content/master/concepts/patch-and-transform.md

```diff
@@ -0,0 +1,1692 @@
+CompositeResourceDefinitions, Claims and EnvironmentConfigs.  
+Only the applied patches change between
+examples. 
+
+All examples rely on Upbound
+[provider-aws-s3](https://marketplace.upbound.io/providers/upbound/provider-aws-s3/)
+to create resources.
+
+{{< expand "Reference Composition" >}}
+```yaml {copy-lines="all"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-composition
+spec:
+  compositeTypeRef:
+    apiVersion: example.org/v1alpha1
+    kind: xExample
+  environment:
+    environmentConfigs:
+    - ref:
+        name: example-environment
+  resources:
+    - name: bucket1
+      base:
+        apiVersion: s3.aws.upbound.io/v1beta1
+        kind: Bucket
+        spec:
+          forProvider:
+            region: us-east-2
+    - name: bucket2
+      base:
+        apiVersion: s3.aws.upbound.io/v1beta1
+        kind: Bucket
+        spec:
+          forProvider:
+            region: us-east-2
+```
+{{< /expand >}}
+
+{{<expand "Reference CompositeResourceDefinition" >}}
+```yaml {copy-lines="all"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: xexamples.example.org
+spec:
+  group: example.org
+  names:
+    kind: xExample
+    plural: xexamples
+  claimNames:
+    kind: ExampleClaim
+    plural: exampleclaims
+  versions:
+  - name: v1alpha1
+    served: true
+    referenceable: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          spec:
+            type: object
+            properties:
+              field1:
+                type: string
+              field2:
+                type: string
+              field3: 
+                type: string
+              desiredRegion: 
+                type: string
+              boolField:
+                type: boolean
+              numberField:
+                type: integer
+          status:
+              type: object
+              properties:
+                url:
+                  type: string
+```
+{{< /expand >}}
+
+
+{{< expand "Reference Claim" >}}
+```yaml {copy-lines="all"}
+apiVersion: example.org/v1alpha1
+kind: ExampleClaim
+metadata:
+  name: my-example-claim
+spec:
+  field1: "field1-text"
+  field2: "field2-text"
+  desiredRegion: "eu-north-1"
+  boolField: false
+  numberField: 10
+```
+{{< /expand >}}
+
+{{< expand "Reference EnvironmentConfig">}}
+```yaml {copy-lines="all"}
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: EnvironmentConfig
+metadata:
+  name: example-environment
+data:
+  locations:
+    us: us-east-2
+    eu: eu-north-1
+  key1: value1
+  key2: value2
+
+```
+{{< /expand >}}
+{{< /hint >}}
+
+<!-- vale Google.Headings = NO -->
+### FromCompositeFieldPath
+<!-- vale Google.Headings = YES -->
+
+The 
+{{<hover label="fromComposite" line="12">}}FromCompositeFieldPath{{</hover>}}
+patch takes a value in a composite resource and applies it to a field in the 
+managed resource. 
+
+{{< hint "tip" >}}
```

Comment:
this is great context to the user about the practical scenario this type of patch is used for